plugins/Installation/FormDatabaseSetup.php

```diff
@@ -57,6 +59,10 @@ function init()
             ->setLabel(Piwik::translate('Installation_DatabaseSetupServer'))
             ->addRule('required', Piwik::translate('General_Required', Piwik::translate('Installation_DatabaseSetupServer')));
 
+        $this->addElement('text', 'port')
+            ->setLabel(Piwik::translate('Installation_DatabaseSetupServerPort'))
+            ->addRule('required', Piwik::translate('General_Required', Piwik::translate('Installation_DatabaseSetupServerPort')));
```

Comment:
After messing with a unix socket for the hostname I am thinking if we need that field at all:

https://github.com/matomo-org/matomo/pull/22252/files#diff-d4f0813ab4aa69d7e57609e4e49758076a032eb18f7d2be481a7489e6b4b2567R181-R185

1. It seems to already work without problems to have the port in the database server, e.g. `127.0.0.1:4000`.
2. I also tried using `DATABASE_HOST=host.docker.internal:4000` in a ddev environment, that worked seemed to work without a problem.
3. To complete my tests I configured myself a matomo-org/docker instance, with `MATOMO_DATABASE_HOST=db:6000` for the app container and `MYSQL_TCP_PORT=6000` for the db container it "just worked".

So it already works, we should not break those ways without intention, adding field that may not be used in the end :thinking:

Reply:
Did that work without the changes on this branch? As the code actually always used 3306 as port in installation.

Reply:
Yes, plain `5.x-dev`.

I started a TiDb on port 4000, and modified a DDEV instance (`.ddev/config.yaml`):

```
omit_containers:
- db
web_environment:
- "DATABASE_HOST=host.docker.internal:4000"
```

The installer displayed the host value including the port and worked without a hitch.

And finding the call to [`Adapter::getDefaultPortForAdapter`](https://github.com/matomo-org/matomo/blob/2743e52dc6c1ca259cc568b1552db304bf661ffb/plugins/Installation/FormDatabaseSetup.php#L133) before the magic [host port socket detection](https://github.com/matomo-org/matomo/blob/2743e52dc6c1ca259cc568b1552db304bf661ffb/plugins/Installation/FormDatabaseSetup.php#L155), it is not unexpected, and should work in all environments alike.

Reply:
I'll rework my PR then. Doesn't make sense to have an additional field then.